### PR TITLE
DMP-2858: Commit to audit for automated task enablement as well as manual execu…

### PR DESCRIPTION
**Pull request by mestebanez**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-2858

### Change description ###

Allow standard and enhanced auditing for enablement/disabled of automated task

Allow standard auditing for manual execution of an automated task

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java [hunk 1]
```diff
@@ -68,6 +74,8 @@ public void runAutomatedTask(Integer taskId) {
         }
 
         automatedTaskRunner.run(automatedTask.get());
+
+        auditApi.record(RUN_JOB_MANUALLY, authorisationApi.getCurrentUser(), null);;
```
Comment (tonymort) [inline src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java:78]: Theres an overloaded api call `record(activity)`that will default to the current user and use null for the case

Comment (mestebanez) [reply src/main/java/uk/gov/hmcts/darts/task/service/impl/AdminAutomatedTasksServiceImpl.java:78]: np. will make the change

File: src/integrationTest/java/uk/gov/hmcts/darts/testutils/EnversAuditQueryHelper.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import jakarta.persistence.Query;
+import org.hibernate.envers.AuditTable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+
+@Component
+public class EnversAuditQueryHelper {
```
[the remark this answers is not in the document]
Comment (mestebanez) [reply src/integrationTest/java/uk/gov/hmcts/darts/testutils/EnversAuditQueryHelper.java:16]: nah just not aware of it. I will change

File: src/main/resources/db/migration/common/V1_313__automated_task_auditing.sql [hunk 3]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO audit_activity (aua_id, activity_name, activity_description, created_ts, created_by, last_modified_ts, last_modified_by)
```
Comment (mestebanez) [inline src/main/resources/db/migration/common/V1_313__automated_task_auditing.sql:1]: Needs correcting

**Review by tonymort — APPROVED** [review]

File: src/main/resources/db/migration/common/V1_312__insert_automated_task_activity.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO audit_activity (aua_id, activity_name, activity_description, created_ts, created_by, last_modified_ts, last_modified_by)
+VALUES (nextval('aua_seq'), 'Run Job Manually', 'Run Job Manually', CURRENT_TIMESTAMP, 0, CURRENT_TIMESTAMP, 0);
+INSERT INTO audit_activity (aua_id, activity_name, activity_description, created_ts, created_by, last_modified_ts, last_modified_by)
+VALUES (nextval('aua_seq'), 'Enable/Disable Job', 'Enable/Disable Job', CURRENT_TIMESTAMP, 0, CURRENT_TIMESTAMP, 0);
```
Comment (jackmaloney) [inline src/main/resources/db/migration/common/V1_312__insert_automated_task_activity.sql:4]: Can you ensure these are added to the modernised_data_standing_data.sql script please?

Comment (mestebanez) [reply src/main/resources/db/migration/common/V1_312__insert_automated_task_activity.sql:4]: np

**Pull request closed without merging**